### PR TITLE
remove qcolor

### PR DESCRIPTION
**Pull request by alexpavlov96**
Resolves: *(direct link to the issue)*

*(short description of the changes and the motivation to make the changes)*

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://github.com/musescore/MuseScore/wiki/CodeGuidelines)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [ ] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [ ] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Is this a replacement for #8663 ?

**Comment by alexpavlov96** [comment]
@Jojo-Schmitz 

> Is this a replacement for #8663 ?
yes, not to handle many conflicts

File: src/engraving/draw/brush.h [hunk 1]
```diff
@@ -32,34 +33,36 @@ class Brush
...
+        Brush brush(Color::fromQColor(qbrush.color()));
         brush.setStyle(static_cast<BrushStyle>(qbrush.style()));
         return brush;
     }
 
 #endif
 
 private:
-    QColor m_color = Qt::black;
+    Color m_color = Ms::MScore::defaultColor;
```
Comment (igorkorsukov) [inline src/engraving/draw/brush.h:65]: I think better just set Color(0,0,0) here

File: src/engraving/draw/color.h [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    int green() const;
+    int blue() const;
+    int alpha() const;
+
+#ifndef NO_QT_SUPPORT
+    static Color fromQColor(const QColor& color);
+    QColor toQColor() const;
+#endif
+
+    static constexpr int m_defaultAlpha = 255;
```
Comment (igorkorsukov) [inline src/engraving/draw/color.h:84]: better rename to `DEFAULT_ALPHA` (this is no regular member, this is constant)

File: src/engraving/draw/painter.h [hunk 3]
```diff
@@ -198,9 +197,9 @@ class Painter
     std::stack<State> m_states;
 };
 
-inline void Painter::setPen(const QColor& color)
+inline void Painter::setPen(const Color& color)
 {
-    setPen(Pen(color.isValid() ? color : QColor(Qt::black)));
+    setPen(Pen(color.isValid() ? color : Ms::MScore::defaultColor));
```
Comment (igorkorsukov) [inline src/engraving/draw/painter.h:202]: better just set `Color(0, 0, 0)`

File: src/engraving/draw/pen.h [hunk 4]
```diff
@@ -39,8 +40,8 @@ class Pen
     {
     }
 
-    Pen(const QColor& color = Qt::black, double width = 1, PenStyle s = PenStyle::SolidLine, PenCapStyle c = PenCapStyle::SquareCap,
-        PenJoinStyle j = PenJoinStyle::BevelJoin)
+    Pen(const Color& color = Ms::MScore::defaultColor, double width = 1, PenStyle s = PenStyle::SolidLine,
```
Comment (igorkorsukov) [inline src/engraving/draw/pen.h:43]: better just set `Color(0, 0, 0)`

File: src/engraving/draw/pen.h [hunk 5]
```diff
@@ -162,8 +163,8 @@ class Pen
 
 private:
 
-    QColor m_color = Qt::black;
-    double m_width = 1;
+    Color m_color = Ms::MScore::defaultColor;
```
Comment (igorkorsukov) [inline src/engraving/draw/pen.h:166]: better just set `Color(0, 0, 0)`

File: src/engraving/draw/utils/drawcomp.cpp [hunk 6]
```diff
@@ -99,7 +99,7 @@ static bool isEqual(const Pen& p1, const Pen& p2)
         return false;
     }
 
-    if (p1.color().name() != p2.color().name()) {
+    if (p1.color().toString() != p2.color().toString()) {
```
Comment (igorkorsukov) [inline src/engraving/draw/utils/drawcomp.cpp:102]: here we can just compare the colors

File: src/engraving/draw/utils/drawcomp.cpp [hunk 7]
```diff
@@ -112,7 +112,7 @@ static bool isEqual(const Brush& b1, const Brush& b2)
         return false;
     }
 
-    if (b1.color().name() != b2.color().name()) {
+    if (b1.color().toString() != b2.color().toString()) {
```
Comment (igorkorsukov) [inline src/engraving/draw/utils/drawcomp.cpp:115]: here we can just compare the colors

File: src/engraving/iengravingconfiguration.h [hunk 8]
```diff
@@ -38,6 +38,25 @@ class IEngravingConfiguration : MODULE_EXPORT_INTERFACE
...
+    virtual QString selectionColor() const = 0;
+    virtual QString warningColor() const = 0;
+    virtual QString warningSelectedColor() const = 0;
+    virtual QString criticalColor() const = 0;
+    virtual QString criticalSelectedColor() const = 0;
+    virtual QString editColor() const = 0;
+    virtual QString harmonyColor() const = 0;
+    virtual QString textBaseFrameColor() const = 0;
+    virtual QString textBaseBgColor() const = 0;
+    virtual QString shadowNoteColor() const = 0;
```
Comment (igorkorsukov) [inline src/engraving/iengravingconfiguration.h:59]: Better to return the Color type right away

File: src/engraving/libmscore/element.cpp [hunk 9]
```diff
@@ -508,17 +508,19 @@ QColor Element::curColor() const
 //   curColor
 //---------------------------------------------------------
 
-QColor Element::curColor(bool isVisible) const
+mu::draw::Color Element::curColor(bool isVisible) const
 {
+    color().toString();
```
Comment (igorkorsukov) [inline src/engraving/libmscore/element.cpp:513]: better remove this )

File: src/engraving/draw/brush.h [hunk 10]
```diff
@@ -22,8 +22,9 @@
 #ifndef MU_DRAW_BRUSH_H
 #define MU_DRAW_BRUSH_H
 
-#include <QColor>
+#include "color.h"
 #include "drawtypes.h"
+#include "libmscore/mscore.h"
```
Comment (igorkorsukov) [inline src/engraving/draw/brush.h:27]: do not forget to remove

File: src/engraving/draw/pen.h [hunk 11]
```diff
@@ -22,12 +22,13 @@
...
 #define MU_DRAW_PEN_H
 
-#include <QColor>
 #ifndef NO_QT_SUPPORT
 #include <QPen>
 #endif
 
+#include "color.h"
 #include "drawtypes.h"
+#include "libmscore/mscore.h"
```
Comment (igorkorsukov) [inline src/engraving/draw/pen.h:31]: do not forget to remove

File: src/engraving/libmscore/figuredbass.cpp [hunk 12]
```diff
@@ -1351,7 +1351,7 @@ void FiguredBass::draw(mu::draw::Painter* painter) const
     if (!score()->printing() && score()->showUnprintable()) {
         for (qreal len : _lineLengths) {
             if (len > 0) {
-                painter->setPen(Pen(Qt::lightGray, 3));
+                painter->setPen(Pen(mu::draw::Color(engravingConfiguration()->figuredBassColor()), 3));
```
Comment (igorkorsukov) [inline src/engraving/libmscore/figuredbass.cpp:1354]: Here and other similar places should be just:
```
painter->setPen(Pen(engravingConfiguration()->figuredBassColor(), 3));
```

**Review by igorkorsukov — APPROVED** [review]

**Pull request merged**